sionna/channel/utils.py

```diff
@@ -65,6 +65,49 @@ def subcarrier_frequencies(num_subcarriers, subcarrier_spacing,
     frequencies = frequencies*subcarrier_spacing
     return frequencies
 
+def time_frequency_vector(num_samples, sample_duration, dtype=tf.float32):
+    # pylint: disable=line-too-long
+    r"""
+    Compute the time and frequency vector for a given number of samples
+    and duration per sample in normalized time unit.
+
+    >>> t = tf.cast(tf.linspace(-n_min, n_max, num_samples), dtype) * sample_duration
+    >>> f = tf.cast(tf.linspace(-n_min, n_max, num_samples), dtype) * 1/(sample_duration*num_samples)
+
+    Input
+    ------
+        num_samples : int
+            Number of samples
+
+        sample_duration : float
+            Sample duration in normalized time
+
+        dtype : tf.DType
+            Datatype to use for internal processing and output.
+            Defaults to `tf.float32`.
+
+    Output
+    ------
+        t : [``num_samples``], ``dtype``
+            Time vector
+
+        f : [``num_samples``], ``dtype``
+            Frequency vector
+    """
+
+    # Time vector
+    n_min = tf.cast(tf.math.ceil((num_samples-1) / 2), dtype=tf.int32)
```

Comment:
I have made some changes to the code so that the actual type was considered. Make sure that I did not break anything here.

Reply:
OK. I added the case where num_samples is odd (by now only even worked).